internal/mapping.go

```diff
@@ -164,3 +164,36 @@ func GetSpannerKeyName(srcKeyName string, schemaForeignKeys map[string]bool) str
 	schemaForeignKeys[spKeyName] = true
 	return spKeyName
 }
+
+// GetSpannerKeyName maps source index key name to
+// legal Spanner index key name
+// We need to make sure
+// of the following things:
+// a) the new index key name is legal
+// b) the new index key name doesn't clash with other Spanner
+//    index key names
+// Note that index key constraint names in Spanner have to be globally unique
+// (across the database). But in some source databases, such as MySQL,
+// they only have to be unique for a table. Hence we must map each source
+// constraint name to a unique spanner constraint name.
+func GetSpannerIndexKeyName(srcKeyName string, schemaIndexKeys map[string]bool) string {
+	spKeyName, _ := FixName(srcKeyName)
+
+	if _, found := schemaIndexKeys[spKeyName]; found {
+		// spKeyName has been used before.
+		// Add unique postfix: use number of index keys so far.
+		// However, there is a chance this has already been used,
+		// so need to iterate.
+		id := len(schemaIndexKeys)
+		for {
+			c := spKeyName + "_" + strconv.Itoa(id)
+			if _, found := schemaIndexKeys[c]; !found {
+				spKeyName = c
+				break
+			}
+			id++
+		}
+	}
```

Comment:
This pattern is now duplicated quite often in the code, consider moving this to a common function 

func findNextUniqueKey(key string, existingKeys) string {
 ...
}

Reply:
separated common code into a function.